pkgs/development/libraries/medfile/default.nix

```diff
@@ -1,19 +1,39 @@
 { lib, stdenv, fetchurl, cmake, hdf5 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "medfile";
-  version = "4.1.1";
+  version = "5.0.0";
 
   src = fetchurl {
-    url = "http://files.salome-platform.org/Salome/other/med-${version}.tar.gz";
-    sha256 = "sha256-3CtdVOvwZm4/8ul0BB0qsNqQYGEyNTcCOrFl1XM4ndA=";
+    url = "https://files.salome-platform.org/Salome/medfile/med-${finalAttrs.version}.tar.bz2";
+    hash = "sha256-Jn520MZ+xRwQ4xmUhOwVCLqo1e2EXGKK32YFKdzno9Q=";
   };
 
-  patches = [
-    ./hdf5-1.14.patch
-  ];
+  outputs = [ "out" "doc" "dev" ];
+
+  postPatch = ''
+    # Patch cmake and source files to work with hdf5
+    substituteInPlace config/cmake_files/medMacros.cmake --replace-fail \
+      "IF (NOT HDF_VERSION_MAJOR_REF EQUAL 1 OR NOT HDF_VERSION_MINOR_REF EQUAL 12 OR NOT HDF_VERSION_RELEASE_REF GREATER 0)" \
+      "IF (HDF5_VERSION VERSION_LESS 1.12.0)"
+    substituteInPlace src/*/*.c --replace-quiet \
+      "#if H5_VERS_MINOR > 12" \
+      "#if H5_VERS_MINOR > 14"
+  '' + lib.optionalString stdenv.isDarwin ''
+    # Some medfile test files #define _a, which
+    # breaks system header files that use _a as a function parameter
+    substituteInPlace tests/c/*.c \
+      --replace-quiet "_a" "_A" \
+      --replace-quiet "_b" "_B"
+    # Fix compiler errors in test files
+    substituteInPlace tests/c/*.c \
+      --replace-quiet "med_Bool" "med_bool" \
+      --replace-quiet "med_Axis_type" "med_axis_type" \
+      --replace-quiet "med_Access_mode" "med_access_mode"
```

Comment:
Shouldn't those be warn or fail? The package might fail when those no longer match.

Reply:
Thanks for pointing this out, @SuperSandro2000, `--replace-warn` is likely the right choice here as `--replace-fail` would fail on files (notice the broad `tests/c/*.c` glob) that do not contain any of the necessary replacements.

I was a bit ~~lazy~~pragmatic in choosing a glob rather than a specific list of files to be replaced, because I thought maintaining that list on future [cw]ould be tedious.

If you see value in being more specific I'll happily make that change…

Reply:
Oh, I totally missed the glob. That makes sense to not use --replace-fail